### ISSUE TITLE
Add method to render a specific block

### ISSUE THREAD

**Issue by Keats** [Keats/tera2#63]
It would render the whole template but only render the content of the requested block.

**Comment by tobiastom** [comment]
Rendering a specific block might still be helpful. I try to use [htmx](https://htmx.org) within one of my projects, and it would be nice to only render the e.g. `content` block. Then I would not return the whole page, but only the content htmx uses. 

I tried to create a htmx component that will capture the content of the rendered template, but as it is not rendered in the context of the template, I cannot get only the inner part of it. 

Do you have any recommendation on how I could implement something like this?

**Comment by Keats** [comment]
To implement it in Tera, you have some fn `Tera::render_block(template_name, block_name)`, you render the template as usual but when you're writing the block you want you also write it to a second buffer and you return that one in the end. I thought with htmx people would use components more, have you seen https://github.com/Keats/tera2/blob/master/MIGRATION.md#components ? If you still want to render a particular block, it shouldn't be too hard to add.

**Comment by tobiastom** [comment]
Yes, I've seen the components and I use them already. They are very cool. Unfortunately I don't see how they would work with htmx. 

When I have a request to e.g. `/search` I collect the data I need and render a template `search.html.tera`. This has multiple blocks I use in an extend. It renders nicely when a normal request is performed.

Now, if the request comes from htmx, I would need to add logic to each handler which component to render. When I change the templates they might diverge. 
That's why I thought the blocks way would be nice convention. If it's htmx I take only the content block, if it's not I take the whole page. No logic needed for each request.

One more thing, I can see that tera has a `render_block` function, tera2 does not seem to have this anymore.

**Comment by Keats** [comment]
> Unfortunately I don't see how they would work with htmx.

I was thinking more fine grained update, eg updating just the navbar component. 

>  I can see that tera has a render_block function

Where?

**Comment by Keats** [comment]
Can you try https://github.com/Keats/tera2/pull/115 ?
